### DISCUSSION TITLE
with `RouterProvider` in Next.js, do i need to wrap `navigate` in `startTransition`?

### DISCUSSION THREAD

**Discussion by stefanprobst** [adobe/react-spectrum#5708] [Q&A]
i am using [`RouterProvider`](https://reactspectrum.blob.core.windows.net/reactspectrum/2ac2748346f665ed7caf3665943593df2db4be66/docs/react-aria/routing.html#app-router) in a next.js app, and i am not sure if the provided `navigate` function (i.e. `router.push`) should be wrapped in `startTransition`?

i had a look what the `next/link` component does internally, and afaiu it *does* wrap `router.push/replace` with `startTransition` when using the app router, see: https://github.com/vercel/next.js/blob/canary/packages/next/src/client/link.tsx#L221-L242

(also not sure if i need to do anything extra for prefetch-on-hover to work?)

thanks!

**Comment by snowystinger** [comment]
See related discussion for more info on prefetch https://github.com/adobe/react-spectrum/issues/5476#issuecomment-1826203368

I'm not sure about the `startTransition`, it seems like we could do that, or since we just call whatever navigate function you provide, you could put in the `startTransition` yourself and wrap whatever you were previously sending in. I don't know if we know enough to always do that.
